src/lib.nr

```diff
@@ -544,3 +545,181 @@ unconstrained fn test_partial_match() {
 
     assert(position == 123);
 }
+
+fn bug_repr(needle: [u8; 32], needle_len: u32, haystack: [u8; 128]) {
+    let id_haystack: StringBody128 = StringBody::new(haystack, 128);
+    let pk_needle: SubString32 = SubString::new(needle, needle_len);
+    let (result, _): (bool, u32) = id_haystack.substring_match(pk_needle);
+    assert(result);
+}
+
+#[test]
+fn bug_test() {
```

Comment:
Is this test pushed intentionally? If so, can be renamed to something meaningful 
Also, I think the `bug_repr` block can be inside the test